common/lib/capa/capa/responsetypes.py

```diff
@@ -252,6 +256,59 @@ def evaluate_answers(self, student_answers, old_cmap):
         # log.debug('new_cmap = %s' % new_cmap)
         return new_cmap
 
+    def make_hint_div(self, hint_node, correct, label=None, hint_text=''):
+        """
+        Given the xml extended hint node and the boolean correctness, returns the extended hint div,
+        like this:
+          <div class="feedback_hint_incorrect">Incorrect: the answer is not 42</div>
+        Optionally, the hint label and the hint_text to use may be passed in,
+        overriding the text found in the hint node, which can be passed as None.
+        This function supports many argument combinations in order to work with the many
+        different contexts where extended hints are constructed.
+        A label with the value '' means inhibit all labeling, while the None means
+        no specific label is provided and we should get the default.
+        The empty string is returned, meaning no hint, if after processing all the arguments,
+        there is no hint.
+        """
+        _ = self.capa_system.i18n.ugettext
+        # Establish the hint text
+        # This can lead to early-exit if the hint is blank.
+        if not hint_text:
+            if hint_node is None or hint_node.text is None:  # .text can be None, maybe just in testing
+                return ''
+            hint_text = hint_node.text.strip()
+            if not hint_text:
+                return ''
+
+        # Establish the label:
+        # Passed in, or from the node, or the default
+        if not label and hint_node is not None:
+            label = hint_node.get('label', None)
+
+        # Tricky: label None means output defaults, while '' means output empty label
+        if label is None:
+            if correct:
+                label = _(u'Correct')
+            else:
+                label = _(u'Incorrect')
+        if label:
+            label += ': '
+
+        # Establish the style
+        if correct:
+            style = QUESTION_HINT_CORRECT_STYLE
+        else:
+            style = QUESTION_HINT_INCORRECT_STYLE
+
+        # Ready to go
+        return u'<div class="{0}">{1}{2}</div>'.format(style, label, hint_text)
+
+    def get_extended_hints(self, student_answers, new_cmap):
+        """
+        Pull "extended hint" information out the xml based on the student answers.
+        Implemented down in the subclasses.
+        """
```

Comment:
Sure. Looking lower in that same class, it seems that "pass" is the thing to do.